libs/libskynet/src/err.ts

```diff
@@ -0,0 +1,47 @@
+import { tryStringify } from "./stringifytry.js";
+
+// addContextToErr is a helper function that standardizes the formatting of
+// adding context to an error. Within the world of go we discovered that being
+// persistent about layering context onto errors is helpful when debugging,
+// even though it often creates rather verbose error messages.
+//
+// addContextToErr will return null if the input err is null.
+//
+// NOTE: To protect against accidental situations where an Error type or some
+// other type is provided instead of a string, we wrap both of the inputs with
+// tryStringify before returning them. This prevents runtime failures.
+function addContextToErr(err: any, context: string): string {
+  if (err === null) {
+    err = "[no error provided]";
+  }
+  err = tryStringify(err);
+  return tryStringify(context) + ": " + tryStringify(err);
+}
+
+// composeErr takes a series of inputs and composes them into a single string.
+// Each element will be separated by a newline. If the input is not a string,
+// it will be transformed into a string with JSON.stringify.
+//
+// Any object that cannot be stringified will be skipped, though an error will
+// be logged.
+function composeErr(...inputs: any): string | null {
```

Comment:
I feel like this whole function reads more complicated than it needs to. There are nice methods on `Array` prototype that can be utilized to achieve the same result. Example:

```ts
function composeErr(...inputs: any): string | null {
  const result = inputs
    .filter(input => input !== null)
    .map(tryStringify)
    .join("\n");
  
  return result || null;  
}
```

Yes, it does more iterations, but is much more readable + the `inputs` array won't be large anyways so it's not worth micro-optimizing here.

Reply:
I'm actually not even sure we use `composeErr` anywhere anymore. It's very useful in `skyd` but in this codebase you don't really have any places where you do 5 actions at once and then check the errors on all 5... javascript is inherently single threaded and I think `composeErr` is more useful in multithreaded environments. I might just toss the function out.